src/components/Carousel.js

```diff
@@ -108,25 +108,18 @@ export default class Carousel extends Component {
     this.resetInterval();
   }
 
-  componentWillReceiveProps(nextProps) {
-    const valueChanged = this.checkIfValueChanged(nextProps);
-
-    if (this.state.transitionEnabled) {
-      return this.setState(previousState => ({
-        transitionEnabled: valueChanged ? true : previousState.transitionEnabled,
-      }));
-    }
-    this.setState(previousState => ({
-      infiniteTransitionFrom: this.getCurrentValue(),
-      transitionEnabled: valueChanged ? true : previousState.transitionEnabled,
-    }));
-  }
-
   componentDidUpdate(prevProps) {
     const valueChanged = this.checkIfValueChanged(prevProps);
     if (this.getProp('autoPlay') !== this.getProp('autoPlay', prevProps) || valueChanged) {
       this.resetInterval();
     }
+
+    if ( valueChanged ) {
```

Comment:
why don't you handle `valueChanged` being false and `this.state.transitionEnabled` being true?

Reply:
The way it looked, the state was being changed only if `valueChanged` was true. Otherwise `previousState` value was being assigned. I'm not sure if it existed for some other reason.
Regarding not handling `this.state.transitionEnabled`, oops! I overlooked it. Thanks for the find.
I'll update RN. The value being set is from `this.getCurrentValue`, which is nothing but `this.props.value`, so going to move it into `getDerivedStateFromProps`. Unit tests are succeeding, kindly tell if I should test anything else.